### PR TITLE
ramda v0.26 add more strict type checking for propEq function

### PR DESCRIPTION
**Pull request by radziksh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint ramda`
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://ramdajs.com/docs/#propEq

### CODE REVIEW

**Comment by radziksh** [comment]
@tycho01 @mdekrey @mrdziuban @sbking @afharo @teves-castro @hojberg @samsonkeung @angeloocana @raynerd @googol @moshensky @ethanresnick @CaptJakk @deftomat @deptno @blimusiek @biern @rayhaneh @rgm @drewwyatt @jottenlips @minitesh @Krantisinh @pirix-gh could you please review my pr when you have time?

**Comment by DanielRosenwasser** [comment]
Ping - this looks good to me, but I wanted to get some feedback from the authors first.

**Review by googol — CHANGES_REQUESTED** [review]
The `= any` defaults to the type parameters seem like a very bad idea to me, it's going to be very easy to forget defining those type parameters

**Comment by radziksh** [comment]
@googol @DanielRosenwasser Yes, this is not an ideal solution, but I had to enter `any` in order to maintain backward compatibility and allow programmers to gradually switch to the typed version of this function. Otherwise, almost all users will have problems with the next update.

**Comment by radziksh** [comment]
@DanielRosenwasser @googol ping

**Comment by googol** [comment]
What is the problem this solves? I am still worried about those `= any`s in the type parameters.

**Comment by radziksh** [comment]
> Ping - @googol @DanielRosenwasser Yes, this is not an ideal solution, but I had to enter `any` in order to maintain backward compatibility and allow programmers to gradually switch to the typed version of this function. Otherwise, almost all users will have problems with the next update.

@googol I wanted to draw attention to my comment, because if I remove `= any` then many users who use this function will have errors after the update. Specifically, on my project, this is not a problem, but I do not want to break the projects of other people, as I see that this library is quite popular. If you still think that any is a bad idea, then I am ready to remove it.

**Comment by googol** [comment]
Right, I understand that. But what is the problem this PR is solving at its core?

**Comment by radziksh** [comment]
Make the world a better place? :)

File: types/ramda/index.d.ts [hunk 1]
```diff
@@ -2530,7 +2531,7 @@ declare namespace R {
          * filter a list.
          */
         propEq<T>(name: string | number, val: T, obj: any): boolean;
-        propEq<T>(name: string | number, val: T): (obj: any) => boolean;
+        propEq<T = any, K extends keyof T = any>(name: K, value: T[K]): (obj: T) => boolean;
```
Comment (Nemo108) [inline types/ramda/index.d.ts]: Не, чувак, так не пойдет. Давай либо все перегрузки правь, либо никакие. И без any, желательно

**Comment by KrzysztofKarol** [comment]
Hi @radziksh, any update on this? It will be very helpfull (type and typo safe)!

**Comment by orta** [comment]
Hey folks, it's been 6 months-ish - the merge conflict on this is pretty wild, and we've not had one approval from someone who owns the definition. I'm going to preemptively close this PR, which doesn't mean that the code is bad, but that we'd like to try this repo fresh given that it's one of the most popular reps in the world. 

Re-opening with fresh code its totally OK 👍

**Pull request closed without merging**